### ISSUE TITLE
Remove tuple types in Repo

### ISSUE THREAD

**Issue by dariusf** [HubTurbo/HubTurbo#1351]
[Class in question](https://github.com/HubTurbo/HubTurbo/blob/master/src/main/java/backend/interfaces/Repo.java). We want to represent results using a(n inner) class instead of a tuple. This would introduce explicit names and make it less fragile (when adding or reordering fields, every implementation and caller wouldn't have to be changed).

This involves creating that class, then updating the interface and all implementations to use it.

**Comment by SherryFX** [comment]
I would like to work on this issue. May I check if creating a public static nested class in the `Repo` interface is okay?

**Comment by dariusf** [comment]
Yup, that's fine. You may need more than one as there are a few different uses of these tuples.

**Comment by SherryFX** [comment]
@dariusf Thanks for the clarification! I would need to replace

**Comment by dariusf** [comment]
Yeah. You might be able to make some of them generic. For example, most of the pairs can be represented by

```java
class Result<T> {
    List<T> contents;
    String eTag;
}
```

or something similar.

**Comment by SherryFX** [comment]
@dariusf Ah I see! Thanks for the tip!
